OpenRA.Mods.Common/Traits/Production.cs

```diff
@@ -1,158 +1,195 @@
-				factionVariant = bi.ForceFaction;
-
-			var td = new TypeDictionary
-			{
-				new OwnerInit(self.Owner),
-			};
-
-			if (self.OccupiesSpace != null)
-			{
-				exit = self.Location + exitinfo.ExitCell;
-				var spawn = self.CenterPosition + exitinfo.SpawnOffset;
-				var to = self.World.Map.CenterOfCell(exit);
-
-				var initialFacing = exitinfo.Facing;
-				if (exitinfo.Facing < 0)
-				{
-					var delta = to - spawn;
-					if (delta.HorizontalLengthSquared == 0)
-					{
-						var fi = producee.TraitInfoOrDefault<IFacingInfo>();
-						initialFacing = fi != null ? fi.GetInitialFacing() : 0;
-					}
-					else
-						initialFacing = delta.Yaw.Facing;
-				}
-
-				exitLocation = rp.Value != null ? rp.Value.Location : exit;
-				target = Target.FromCell(self.World, exitLocation);
-
-				td.Add(new LocationInit(exit));
-				td.Add(new CenterPositionInit(spawn));
-				td.Add(new FacingInit(initialFacing));
-			}
-
-			self.World.AddFrameEndTask(w =>
-			{
-				if (factionVariant != null)
-					td.Add(new FactionInit(factionVariant));
-
-				var newUnit = self.World.CreateActor(producee.Name, td);
-
-				var move = newUnit.TraitOrDefault<IMove>();
-				if (move != null)
-				{
-					if (exitinfo.MoveIntoWorld)
-					{
-						if (exitinfo.ExitDelay > 0)
-							newUnit.QueueActivity(new Wait(exitinfo.ExitDelay, false));
-
-						newUnit.QueueActivity(move.MoveIntoWorld(newUnit, exit));
-						newUnit.QueueActivity(new AttackMoveActivity(
-							newUnit, move.MoveTo(exitLocation, 1)));
-					}
-				}
-
-				newUnit.SetTargetLine(target, rp.Value != null ? Color.Red : Color.Green, false);
-
-				if (!self.IsDead)
-					foreach (var t in self.TraitsImplementing<INotifyProduction>())
-						t.UnitProduced(self, newUnit, exit);
-
-				var notifyOthers = self.World.ActorsWithTrait<INotifyOtherProduction>();
-				foreach (var notify in notifyOthers)
-					notify.Trait.UnitProducedByOther(notify.Actor, self, newUnit);
-
-				foreach (var t in newUnit.TraitsImplementing<INotifyBuildComplete>())
-					t.BuildingComplete(newUnit);
-			});
-		}
-
-		public virtual bool Produce(Actor self, ActorInfo producee, string factionVariant)
-		{
-			if (Reservable.IsReserved(self) || (building != null && building.Locked))
-				return false;
-
-			// Pick a spawn/exit point pair
-			var exit = self.Info.TraitInfos<ExitInfo>().Shuffle(self.World.SharedRandom)
-				.FirstOrDefault(e => CanUseExit(self, producee, e));
-
-			if (exit != null || self.OccupiesSpace == null)
-			{
-				DoProduction(self, producee, exit, factionVariant);
-				return true;
-			}
-
-			return false;
-		}
-
-		static bool CanUseExit(Actor self, ActorInfo producee, ExitInfo s)
-		{
-			var mobileInfo = producee.TraitInfoOrDefault<MobileInfo>();
-
-			self.NotifyBlocker(self.Location + s.ExitCell);
-
-			return mobileInfo == null ||
-				mobileInfo.CanEnterCell(self.World, self, self.Location + s.ExitCell, self);
-		}
-	}
-}
+#region Copyright & License Information

```

Comment:
Likewise here: please keep the line endings as they were.

Reply:
Hmm, VS doesn't report problem there but I will check again.

Reply:
In VS you can use File-Advanced Save Options to revert this per file. VS does tend to use CRLF endings even in an LF file.

Reply:
Advanced Save is only available VS2016 and lower. But no worries, I did it in N++ already so it should be in next push.